compiler/compiler/src/run_with_ledger.rs

```diff
@@ -0,0 +1,299 @@
+        .expect("Failed to prepare advance to next beacon block");
+    // Assert that no transactions were aborted or rejected.
+    assert!(block.aborted_transaction_ids().is_empty());
+    assert_eq!(block.transactions().num_rejected(), 0);
+    // Advance the ledger to the next block.
+    ledger.advance_to_next_block(&block).expect("Failed to advance to next block");
+
+    // Advance the ledger with empty blocks until the specified height.
+    let current_height = ledger.vm().block_store().current_block_height();
+    let num_blocks = config.min_height.saturating_sub(current_height).saturating_sub(1);
+    for _ in 0..num_blocks {
+        let block = ledger
+            .prepare_advance_to_next_beacon_block(&genesis_private_key, vec![], vec![], vec![], &mut rng)
+            .expect("Failed to prepare advance to next beacon block");
+        ledger.advance_to_next_block(&block).expect("Failed to advance to next block");
+    }
+
+    let mut case_outcomes = Vec::new();
+
+    for case in cases {
+        assert!(
+            ledger.vm().contains_program(&ProgramID::from_str(&case.program_name).unwrap()),
+            "Program {} should exist.",
+            case.program_name
+        );
+
+        let private_key = case
+            .private_key
+            .as_ref()
+            .map(|key| PrivateKey::from_str(key).expect("Failed to parse private key."))
+            .unwrap_or(genesis_private_key);
+
+        let mut execution = None;
+        let mut verified = false;
+        let mut status = Status::None;
+
+        // Halts are handled by panics, so we need to catch them.
+        // I'm not thrilled about this usage of `AssertUnwindSafe`, but it seems to be
+        // used frequently in SnarkVM anyway.
+        let execute_output = std::panic::catch_unwind(std::panic::AssertUnwindSafe(|| {
+            ledger.vm().execute(
+                &private_key,
+                (&case.program_name, &case.function),
+                case.input.iter(),
+                None,
+                0,
+                None,
+                &mut rng,
+            )
+        }));
+
+        if let Err(payload) = execute_output {
+            let s1 = payload.downcast_ref::<&str>().map(|s| s.to_string());
+            let s2 = payload.downcast_ref::<String>().cloned();
+            let s = s1.or(s2).unwrap_or_else(|| "Unknown panic payload".to_string());
+
+            case_outcomes.push(CaseOutcome {
+                status: Status::Halted(s),
+                verified: false,
+                errors: buf.extract_errs(),
+                warnings: buf.extract_warnings(),
+                execution: "".to_string(),
+            });
+            continue;
+        }
+
+        let result = execute_output
+            .unwrap()
+            .and_then(|transaction| {
+                verified = ledger.vm().check_transaction(&transaction, None, &mut rng).is_ok();
```

Comment:
Would we want to change `verified` to a `String` so that we can bubble up the error message?